CMakeLists.txt

```
@@ -224,6 +229,7 @@ if (BASE64_WITH_OpenMP)
         PRIVATE
             ${OpenMP_C_FLAGS}
     )
+    target_link_libraries(base64 OpenMP::OpenMP_C)
```

This is incompatible with our current minimum supported CMake version which is `3.1`. Personally I have no problems raising it, but I think we should make a concious decision here based on what CMake versions are available in widely used distributions like RHEL, ubuntu LTS, etc.
cc: @madebr

---

Minimum CMake version providing `OpenMP::OpenMP_C` is 3.9:
https://cmake.org/cmake/help/v3.9/module/FindOpenMP.html
This one was released in [July 2017](https://cmake.org/pipermail/cmake/2017-July/065814.html)

3.9 is available in Ubuntu 18.04 LTS (which will be EOL next year): https://launchpad.net/ubuntu/bionic/+source/cmake

Personally, I'm completely fine with bumping `cmake_minimum_required`.

small suggestion:
```suggestion
    target_link_libraries(base64 PRIVATE OpenMP::OpenMP_C)
```

---

Using [this github action](https://github.com/marketplace/actions/actions-setup-cmake), you can enforce a specific cmake version to ensure compatibility with older versions.